src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentPreprocessorDirectiveTest.cs

```diff
@@ -0,0 +1,727 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Razor.Language.IntegrationTests;
+
+public class ComponentPreprocessorDirectiveTest(bool designTime = false)
+        : RazorBaselineIntegrationTestBase(layer: TestProject.Layer.Compiler)
+{
+    internal override string FileKind => FileKinds.Component;
+
+    internal override bool UseTwoPhaseCompilation => true;
+
+    internal string ComponentName = "TestComponent";
+
+    internal override string DefaultFileName => ComponentName + ".cshtml";
```

Comment:
I know we probably have this in other "component" tests, but shouldn't components have file extension `.razor` instead of `.cshtml`?

Reply:
Probably.